repo-sync-2025-01-02T18:56:23+0800

w-gc opened the pull request:
- [SPU] Migrating to Bazel Modules and update minimum Python version to 3.10.
- [Feature] Add soPRF (LowMC cipher) for SEMI2K
- [API] Add Permute/InvPermute support in HLO
- [Feature] Add SSL configuration to the TTP server
- [Feature] Support quick sort for semi2k and aby3

tpppppub commented:
CONTRIBUING.md (maybe more docs) and circleci scripts should also be updated. Besides, I don't get why drop python3.9 at this moment as it's not beyond the end of life.

tpppppub commented:
The other pending PRs may also be notified this change. It will be better to merge them first as this PR will cause a huge amount of conflicts.

w-gc commented:
> CONTRIBUING.md (maybe more docs) and circleci scripts should also be updated. Besides, I don't get why drop python3.9 at this moment as it's not beyond the end of life.

@tpppppub Okay, I will update them. Python 3.9 is out of the supported window of Python and JAX now. And, it's a good time to update minimum Python version to 3.10, because we have just successfully released SPU 0.9.3b, and we are now migrating to Bazel Modules, which is a big change.

ref:
- [SPEC 0 — Minimum Supported Dependencies](https://scientific-python.org/specs/spec-0000/?spm=137c942b.3083ba24.0.0.19fcegzIegzINe)
- [JAX Python and NumPy version support policy](https://jax.readthedocs.io/en/latest/deprecation.html?spm=137c942b.3083ba24.0.0.19fcegzIegzINe)

tongke6 reviewed (approved):
LGTM

tongke6 commented on bazel/patches/grpc-1.66.patch:
```diff
@@ -0,0 +1,19 @@
+--- a/third_party/BUILD
++++ b/third_party/BUILD
+@@ -18,13 +18,13 @@ package(default_visibility = ["//:__subpackages__"])
+
```
the patch file is not correctly synced, the last empty line missed, see diff with see https://github.com/secretflow/psi/blob/main/bazel/patches/grpc-1.66.patch

w-gc replied:
FIXED

tongke6 reviewed (commented):
requirements-dev.txt and requirements.txt are still needed by //:spu_wheel and //:requirements-dev.update commands

w-gc commented:
> The other pending PRs may also be notified this change. It will be better to merge them first as this PR will cause a huge amount of conflicts.

I have check the conflicts about branch `w-gc:sync` and pending PRs.  The conflicts are small than we feared：
```bash
---------------start to check PR952---------------
---------------start to check PR952---------------
Already on 'main'
Your branch is up to date with 'origin/main'.
From https://github.com/secretflow/spu
 * [new ref]         refs/pull/952/head -> PR952
Switched to branch 'PR952'
Auto-merging sml/decomposition/BUILD.bazel
Automatic merge went well; stopped before committing as requested
---------------start to check PR951---------------
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
error: branch 'PR951' not found
From https://github.com/secretflow/spu
 * [new ref]         refs/pull/951/head -> PR951
Switched to branch 'PR951'
Auto-merging libspu/kernel/hal/permute.h
CONFLICT (content): Merge conflict in libspu/kernel/hal/permute.h
Auto-merging sml/manifold/BUILD.bazel
CONFLICT (content): Merge conflict in sml/manifold/BUILD.bazel
Auto-merging spu/intrinsic/BUILD.bazel
Automatic merge failed; fix conflicts and then commit the result.
---------------start to check PR950---------------
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
From https://github.com/secretflow/spu
 * [new ref]         refs/pull/950/head -> PR950
Switched to branch 'PR950'
Auto-merging libspu/version.h.in
CONFLICT (content): Merge conflict in libspu/version.h.in
Automatic merge failed; fix conflicts and then commit the result.
---------------start to check PR939---------------
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
From https://github.com/secretflow/spu
 * [new ref]         refs/pull/939/head -> PR939
Switched to branch 'PR939'
Automatic merge went well; stopped before committing as requested
---------------start to check PR881---------------
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
From https://github.com/secretflow/spu
 * [new ref]         refs/pull/881/head -> PR881
Switched to branch 'PR881'
fatal: refusing to merge unrelated histories
fatal: There is no merge to abort (MERGE_HEAD missing).
---------------start to check PR880---------------
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
From https://github.com/secretflow/spu
 * [new ref]         refs/pull/880/head -> PR880
Switched to branch 'PR880'
Auto-merging libspu/core/config.cc
Auto-merging libspu/core/encoding.cc
Auto-merging libspu/mpc/BUILD.bazel
Auto-merging libspu/mpc/ab_api_test.cc
Auto-merging libspu/mpc/api.cc
Auto-merging libspu/mpc/api.h
Auto-merging libspu/mpc/common/BUILD.bazel
Auto-merging libspu/mpc/common/communicator.cc
CONFLICT (content): Merge conflict in libspu/mpc/common/communicator.cc
Auto-merging libspu/mpc/common/communicator.h
CONFLICT (content): Merge conflict in libspu/mpc/common/communicator.h
Auto-merging libspu/mpc/kernel.cc
Auto-merging libspu/mpc/kernel.h
Auto-merging libspu/mpc/utils/BUILD.bazel
CONFLICT (content): Merge conflict in libspu/mpc/utils/BUILD.bazel
Auto-merging libspu/mpc/utils/gfmp.h
CONFLICT (add/add): Merge conflict in libspu/mpc/utils/gfmp.h
Auto-merging libspu/mpc/utils/gfmp_ops.cc
CONFLICT (add/add): Merge conflict in libspu/mpc/utils/gfmp_ops.cc
Auto-merging libspu/mpc/utils/gfmp_ops.h
CONFLICT (add/add): Merge conflict in libspu/mpc/utils/gfmp_ops.h
Auto-merging libspu/spu.proto
Automatic merge failed; fix conflicts and then commit the result.
---------------start to check PR842---------------
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
From https://github.com/secretflow/spu
 * [new ref]         refs/pull/842/head -> PR842
Switched to branch 'PR842'
Automatic merge went well; stopped before committing as requested
```

tongke6 commented:
> > The other pending PRs may also be notified this change. It will be better to merge them first as this PR will cause a huge amount of conflicts.
> 
> I have checked the conflicts about branch `w-gc:sync` and pending PRs. The conflicts are small than we feared：
> 
> ```shell
> ---------------start to check PR952---------------
> Already on 'main'
> Your branch is up to date with 'origin/main'.
> From https://github.com/secretflow/spu
>  * [new ref]         refs/pull/952/head -> PR952
> Switched to branch 'PR952'
> Auto-merging sml/decomposition/BUILD.bazel
> Automatic merge went well; stopped before committing as requested
> ---------------start to check PR951---------------
> Switched to branch 'main'
> Your branch is up to date with 'origin/main'.
> error: branch 'PR951' not found
> From https://github.com/secretflow/spu
>  * [new ref]         refs/pull/951/head -> PR951
> Switched to branch 'PR951'
> Auto-merging libspu/kernel/hal/permute.h
> CONFLICT (content): Merge conflict in libspu/kernel/hal/permute.h
> Auto-merging sml/manifold/BUILD.bazel
> CONFLICT (content): Merge conflict in sml/manifold/BUILD.bazel
> Auto-merging spu/intrinsic/BUILD.bazel
> Automatic merge failed; fix conflicts and then commit the result.
> ---------------start to check PR950---------------
> Switched to branch 'main'
> Your branch is up to date with 'origin/main'.
> From https://github.com/secretflow/spu
>  * [new ref]         refs/pull/950/head -> PR950
> Switched to branch 'PR950'
> Auto-merging libspu/version.h.in
> CONFLICT (content): Merge conflict in libspu/version.h.in
> Automatic merge failed; fix conflicts and then commit the result.
> ---------------start to check PR939---------------
> Switched to branch 'main'
> Your branch is up to date with 'origin/main'.
> From https://github.com/secretflow/spu
>  * [new ref]         refs/pull/939/head -> PR939
> Switched to branch 'PR939'
> Automatic merge went well; stopped before committing as requested
> ---------------start to check PR881---------------
> Switched to branch 'main'
> Your branch is up to date with 'origin/main'.
> From https://github.com/secretflow/spu
>  * [new ref]         refs/pull/881/head -> PR881
> Switched to branch 'PR881'
> fatal: refusing to merge unrelated histories
> fatal: There is no merge to abort (MERGE_HEAD missing).
> ---------------start to check PR880---------------
> Switched to branch 'main'
> Your branch is up to date with 'origin/main'.
> From https://github.com/secretflow/spu
>  * [new ref]         refs/pull/880/head -> PR880
> Switched to branch 'PR880'
> Auto-merging libspu/core/config.cc
> Auto-merging libspu/core/encoding.cc
> Auto-merging libspu/mpc/BUILD.bazel
> Auto-merging libspu/mpc/ab_api_test.cc
> Auto-merging libspu/mpc/api.cc
> Auto-merging libspu/mpc/api.h
> Auto-merging libspu/mpc/common/BUILD.bazel
> Auto-merging libspu/mpc/common/communicator.cc
> CONFLICT (content): Merge conflict in libspu/mpc/common/communicator.cc
> Auto-merging libspu/mpc/common/communicator.h
> CONFLICT (content): Merge conflict in libspu/mpc/common/communicator.h
> Auto-merging libspu/mpc/kernel.cc
> Auto-merging libspu/mpc/kernel.h
> Auto-merging libspu/mpc/utils/BUILD.bazel
> CONFLICT (content): Merge conflict in libspu/mpc/utils/BUILD.bazel
> Auto-merging libspu/mpc/utils/gfmp.h
> CONFLICT (add/add): Merge conflict in libspu/mpc/utils/gfmp.h
> Auto-merging libspu/mpc/utils/gfmp_ops.cc
> CONFLICT (add/add): Merge conflict in libspu/mpc/utils/gfmp_ops.cc
> Auto-merging libspu/mpc/utils/gfmp_ops.h
> CONFLICT (add/add): Merge conflict in libspu/mpc/utils/gfmp_ops.h
> Auto-merging libspu/spu.proto
> Automatic merge failed; fix conflicts and then commit the result.
> ---------------start to check PR842---------------
> Switched to branch 'main'
> Your branch is up to date with 'origin/main'.
> From https://github.com/secretflow/spu
>  * [new ref]         refs/pull/842/head -> PR842
> Switched to branch 'PR842'
> Automatic merge went well; stopped before committing as requested
> ```

Agree +1

tongke6 reviewed (approved):
LGTM
